src/encoded/static/components/rnaget.js

```diff
@@ -0,0 +1,889 @@
+ * Displays the image download_url path as a link to the image.
+ */
+const ImageDownloadRenderer = ({ value }, reactContext) => {
+    const parsedUrl = url.parse(reactContext.location_href);
+    return <a href={`${parsedUrl.protocol}//${parsedUrl.host}${value}`}>{value}</a>;
+};
+
+ImageDownloadRenderer.propTypes = {
+    /** path to image */
+    value: PropTypes.string.isRequired,
+};
+
+ImageDownloadRenderer.contextTypes = {
+    location_href: PropTypes.string,
+};
+
+RNAGetDataRegistry.register({ field: 'download_url', type: 'Image' }, ImageDownloadRenderer);
+
+
+/**
+ * `files` renderer for type=Experiment. Displays a comma separated list of accessions that link to
+ * the corresponding file page.
+ */
+const ExperimentFilesRenderer = ({ item }) => {
+    if (item.files && item.files.length > 0) {
+        return item.files.map((file) => <a key={file['@id']} href={file['@id']}>{globals.atIdToAccession(file['@id'])}</a>).reduce((prev, curr) => [prev, ', ', curr]);
+    }
+    return null;
+};
+
+ExperimentFilesRenderer.propTypes = {
+    /** Object (row) currently rendered */
+    item: PropTypes.object.isRequired,
+};
+
+RNAGetDataRegistry.register({ field: 'files.@id', type: 'Experiment' }, ExperimentFilesRenderer);
+
+
+/**
+ * Make complete URLs from the file download paths.
+ */
+const FileDownloadHref = ({ value }, reactContext) => {
+    const parsedUrl = url.parse(reactContext.location_href);
+    return `${parsedUrl.protocol}//${parsedUrl.host}${value}`;
+};
+
+FileDownloadHref.contextTypes = {
+    location_href: PropTypes.string,
+};
+
+RNAGetDataRegistry.register({ field: 'href', type: 'File' }, FileDownloadHref);
+
+
+/**
+ * Get the property specified by a field from the search results and return a string from that.
+ */
+const lookupColumn = (result, column) => {
+    let nodes = [result];
+    const names = column.split('.');
+
+    for (let i = 0, len = names.length; i < len && nodes.length > 0; i += 1) {
+        let nextnodes = [];
+        _.each(nodes.map((node) => node[names[i]]), (v) => {
+            if (v === undefined) return;
+            if (Array.isArray(v)) {
+                nextnodes = nextnodes.concat(v);
+            } else {
+                nextnodes.push(v);
+            }
+        });
+        if (names[i + 1] === 'length' || names[i + 1] === 'uuid') {
+            // Displaying the length of an array. That's not a property of each array element so we
+            // can't get it that way. Just return the length of the array.
+            nodes = [nextnodes.length];
+            break;
+        } else {
+            // Moving on to the next node defined by the `names` array.
+            nodes = nextnodes;
+        }
+    }
+    // if we ended with an embedded object, show the @id
+    if (nodes.length > 0 && nodes[0]['@id'] !== undefined) {
+        nodes = nodes.map((node) => node['@id']);
+    }
+
+    // Stringify any nodes that are objects or arrays. Objects and arrays have typeof `object`.
```

Comment:
"Stringify any nodes that are **collections**."

Everyone knows most things in Javascript are objects. The CEO of Brave-browser who designed Javascript was obsessed with them. So there is no need to tell developers.